src/index.js

```diff
@@ -85,6 +86,18 @@ module.exports = function SitemapGenerator(uri, opts) {
     ) {
       emitter.emit('ignore', url);
     } else {
+      // https://zendesk.atlassian.net/browse/WT-5268 - ignore canonicalized pages
+      if (options.ignoreCanonacalized) {
+        const canonicalMatches = /<link rel="canonical" href="([^"]*)"/gi.exec(page);
+        if (canonicalMatches && canonicalMatches.length > 1) {
```

Comment:
This ensures that we are only excluding a page if it _has_ a canonical defined and doesn't match, versus just excluding every page without a canonical.